src/promptflow-azure/promptflow/azure/_utils/_tracing.py

```diff
@@ -53,20 +90,24 @@ def validate_trace_provider(value: str) -> None:
         ml_client.workspaces.get(name=workspace_triad.workspace_name)
     except ResourceNotFoundError as e:
         raise _create_trace_provider_value_user_error(str(e))
+    _logger.debug("Azure ML workspace is valid.")
 
     # the workspace Cosmos DB is initialized
-    # call PFS API to try to retrieve the token
-    # otherwise, print the trace ui and hint the user to init the Cosmos DB from Azure portal
+    # try to retrieve the token from PFS; if failed, call PFS init API and start polling
+    _logger.debug("Validating workspace Cosmos DB is initialized...")
     pf_client = PFClient(ml_client=ml_client)
     try:
         pf_client._traces._get_cosmos_db_token(container_name=CosmosDBContainerName.SPAN)
-    except FlowRequestException as e:
-        ws_tracing_url = get_ws_tracing_base_url(workspace_triad)
-        msg = (
-            f"Failed attempt to retrieve the Cosmos DB token: {str(e)}, "
-            "this might because you have not initialized the Cosmos DB for the given workspace, "
-            "or it's still be initializing.\n"
-            f"Please open the following link to manually initialize it: {ws_tracing_url}; "
-            "when it's done, retry the command to set the trace provider again."
+        _logger.debug("The workspace Cosmos DB is already initialized.")
+    except FlowRequestException:
```

Comment:
Is that possible to check the detail error info?
Seems like we will start to initialize for all FlowRequestException exception. (Say if I don't have permission for the ws, we will try to init also)

Reply:
to be honest, i never run into error requesting to PFS... and for your case (i know it's just an example), at the 2nd step, we will validate permission by getting workspace via v2 SDK, that will somehow validate the permission... in short, let's keep this and see if users will raise some.

Reply:
Traceback (most recent call last):
  File "/home/robbenwang/SourceCode/test_trace/test_trace.py", line 3, in <module>
    get_client("test", "96aede12-2f73-41cb-b983-6d11a904839b", "robbenwang-test-rg", "robbenwang-scope-test-ws-westus2")
  File "/home/robbenwang/SourceCode/promptflow/src/promptflow-azure/promptflow/azure/_storage/cosmosdb/client.py", line 35, in get_client
    token = _get_resource_token(
  File "/home/robbenwang/SourceCode/promptflow/src/promptflow-azure/promptflow/azure/_storage/cosmosdb/client.py", line 88, in _get_resource_token
    token_resp = pf_client._traces._get_cosmos_db_token(container_name=container_name, acquire_write=True)
  File "/home/robbenwang/SourceCode/promptflow/src/promptflow-devkit/promptflow/_sdk/_telemetry/activity.py", line 265, in wrapper
    return f(self, *args, **kwargs)
  File "/home/robbenwang/SourceCode/promptflow/src/promptflow-azure/promptflow/azure/operations/_trace_operations.py", line 34, in _get_cosmos_db_token
    return self._service_caller.get_cosmos_resource_token(
  File "/home/robbenwang/SourceCode/promptflow/src/promptflow-azure/promptflow/azure/_restclient/flow_service_caller.py", line 65, in wrapper
    raise FlowRequestException(
promptflow.azure._restclient.flow_service_caller.FlowRequestException: Calling get_cosmos_resource_token failed with request id: 4ed2f78d-953d-4782-991b-ae523afc4fda 
**Status code: 404** 
Reason: Not found trace session database for workspace robbenwang-scope-test-ws-westus2 
Error message: (UserError) Not found trace session database for workspace robbenwang-scope-test-ws-westus2
Code: UserError
**Message: Not found trace session database for workspace robbenwang-scope-test-ws-westus2** 

Maybe we could use specific error code or error message to make decision about creating cosmosDB?

Reply:
good catch, have an offline sync with Robben, we can layer this deeper later.